runtime/src/main/kotlin/kotlin/native/concurrent/Continuation.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package kotlin.native.concurrent
+
+import kotlin.native.internal.*
+import kotlinx.cinterop.*
+
+public class Continuation0(block: () -> Unit,
+                    private val invoker: CPointer<CFunction<(COpaquePointer?) -> Unit>>,
```

Comment:
What's the purpose of using pointers in API here? I believe this class can take frozen lambda as invoker.
So invoker can be like `(() -> Unit) -> Unit` or even
```
interface Invoker {
    fun invoke(block: () -> Unit)
}
```

The invoker gets shared wrapper for the original `block` with its arguments bound, and can invoke it directly from appropriate context. This eliminates the requirement to have complicated work-flow with `callContinuation*`.

Reply:
Interesting point. It was done that way as practical invokers I was planning for are ones that call C/Obj-C APIs and pass C callbacks and `void*` argument, so API kind of modelled around this pattern.

Reply:
Ok, now I see. I suppose we can keep this base version, but return back later to improve support for block-style APIs.